arangod/V8Server/v8-actions.cpp

```diff
@@ -436,6 +436,62 @@ v8::Handle<v8::Object> TRI_RequestCppToV8(v8::Isolate* isolate,
   // intentional copy, as we will modify the headers later
   auto headers = request->headers();
 
+  switch (request->acceptEncoding()) {
+  case EncodingType::UNSET:
+    headers.insert(std::pair<std::string, std::string>(StaticStrings::AcceptEncoding, StaticStrings::EncodingIdentity));
+    break;
+  case EncodingType::DEFLATE:
+    headers.insert(std::pair<std::string, std::string>(StaticStrings::AcceptEncoding, StaticStrings::EncodingDeflate));
+    break;
+  }
+  std::string const& acceptPlain = request->contentTypeResponsePlain();
+
+  if (!acceptPlain.empty()) {
+    headers.insert(std::pair<std::string, std::string>(StaticStrings::Accept, acceptPlain));
+  } else {
+    switch(request->contentTypeResponse()) {
+    case ContentType::UNSET:
+    case ContentType::CUSTOM:  // use Content-Type from _headers
+      break;
+    case ContentType::JSON:    // application/json
+      headers.insert(std::pair<std::string, std::string>(StaticStrings::Accept, StaticStrings::MimeTypeJson));
+      break;
+    case ContentType::VPACK:   // application/x-velocypack
+      headers.insert(std::pair<std::string, std::string>(StaticStrings::Accept, StaticStrings::MimeTypeVPack));
+      break;
+    case ContentType::TEXT:    // text/plain
+      headers.insert(std::pair<std::string, std::string>(StaticStrings::Accept, StaticStrings::MimeTypeText));
+      break;
+    case ContentType::HTML:    // text/html
+      headers.insert(std::pair<std::string, std::string>(StaticStrings::Accept, StaticStrings::MimeTypeHtml));
+      break;
+    case ContentType::DUMP:    // application/x-arango-dump
+      headers.insert(std::pair<std::string, std::string>(StaticStrings::Accept, StaticStrings::MimeTypeDump));
+      break;
+    }
+  }
+
+  switch(request->contentType()) {
+    case ContentType::UNSET:
+    case ContentType::CUSTOM:  // use Content-Type from _headers
+      break;
+    case ContentType::JSON:    // application/json
+      headers.insert(std::pair<std::string, std::string>(StaticStrings::ContentTypeHeader, StaticStrings::MimeTypeJson));
+      break;
+    case ContentType::VPACK:   // application/x-velocypack
+      headers.insert(std::pair<std::string, std::string>(StaticStrings::ContentTypeHeader, StaticStrings::MimeTypeVPack));
+      break;
+    case ContentType::TEXT:    // text/plain
+      headers.insert(std::pair<std::string, std::string>(StaticStrings::ContentTypeHeader, StaticStrings::MimeTypeText));
+      break;
+    case ContentType::HTML:    // text/html
+      headers.insert(std::pair<std::string, std::string>(StaticStrings::ContentTypeHeader, StaticStrings::MimeTypeHtml));
+      break;
+    case ContentType::DUMP:    // application/x-arango-dump
+      headers.insert(std::pair<std::string, std::string>(StaticStrings::ContentTypeHeader, StaticStrings::MimeTypeDump));
+      break;
```

Comment:
Instead of `map::insert(std::pair<std::string, std::string>(key, value))`, can't we simply use `map::emplace(key, value`?